CMakeLists.txt

```diff
@@ -203,18 +203,20 @@ else()
 endif()
 
 include(CheckCCompilerFlag)
+include(CheckCXXSourceCompiles)
 if(CMAKE_SYSTEM_PROCESSOR MATCHES "^(powerpc|ppc)64")
-  CHECK_C_COMPILER_FLAG("-mcpu=power9" HAS_POWER9)
-  if(HAS_POWER9)
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mcpu=power9 -mtune=power9")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mcpu=power9 -mtune=power9")
-  else()
-    CHECK_C_COMPILER_FLAG("-mcpu=power8" HAS_POWER8)
-    if(HAS_POWER8)
-      set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mcpu=power8 -mtune=power8")
-      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mcpu=power8 -mtune=power8")
-    endif(HAS_POWER8)
-  endif(HAS_POWER9)
+  CHECK_CXX_SOURCE_COMPILES("
+  #ifndef _ARCH_PWR8
+  #error Not POWER8+
+  #endif
+  int main() { return 0; }"
+  HAS_POWER8)
```

Comment:
If I understand this correctly `CHECK_CXX_SOURCE_COMPILES` sets the `HAS_POWER8` flag only if the machine support the POWER 8 (or higher) ISA... however it then appears that this flag is not used anywhere else?